provider/kubernetes/kubernetes_test.go

```diff
@@ -1224,6 +1224,24 @@ rateset:
 					iPaths(onePath(iPath("/customheaders"), iBackend("service1", intstr.FromInt(80))))),
 			),
 		),
+		buildIngress(
+			iNamespace("testing"),
+			iAnnotation(annotationKubernetesProtocol, "h2c"),
+			iRules(
+				iRule(
+					iHost("protocol"),
+					iPaths(onePath(iPath("/valid"), iBackend("service1", intstr.FromInt(80))))),
+			),
+		),
```

Comment:
How about adding the test case where we don't use port 443 (i.e., protocol is set to `http`) but explicitly specify the `https` protocol through the annotation?

Reply:
I added some more tests:)